### PR TITLE
Convert GetCurrentMethod to QCALL

### PR DESCRIPTION
**Pull request by AustinWise**
This has a similar structure to `FailFast` as converted in #98908.

Contributes to #95695

### CODE REVIEW

File: src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs [hunk 1]
```diff
@@ -844,11 +844,13 @@ public IntPtr GetFunctionPointer()
...
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern IRuntimeMethodInfo? _GetCurrentMethod(ref StackCrawlMark stackMark);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "RuntimeMethodHandle_GetCurrentMethod")]
+        private static partial void _GetCurrentMethod(StackCrawlMarkHandle stackMark, ObjectHandleOnStack retMethod);
         internal static IRuntimeMethodInfo? GetCurrentMethod(ref StackCrawlMark stackMark)
         {
-            return _GetCurrentMethod(ref stackMark);
+            object? retMethod = null;
+            _GetCurrentMethod(new StackCrawlMarkHandle(ref stackMark), ObjectHandleOnStack.Create(ref retMethod));
+            return (IRuntimeMethodInfo?)retMethod;
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs:853]:
```suggestion
            IRuntimeMethodInfo? retMethod = null;
            _GetCurrentMethod(new StackCrawlMarkHandle(ref stackMark), ObjectHandleOnStack.Create(ref retMethod));
            return retMethod;
```

**Comment by jkotas** [comment]
I see about 6% perf regression in `MethodBase.GetCurrentMethod` with this change (Win x64). 

```
using System.Reflection;
using System.Diagnostics;

for(;;)
{
    Stopwatch sw = new Stopwatch();
    sw.Start();
    for (int i = 0; i < 1000000; i++) MethodBase.GetCurrentMethod();
    Console.WriteLine(sw.ElapsedMilliseconds.ToString());
}
```

The implementation of `MethodBase.GetCurrentMethod` comes with a lot of unnecessary ceremony. Eliminating the ceremony should fix the regression - it can be just:

```
        public static MethodBase? GetCurrentMethod()
        {
            StackCrawlMark stackMark = StackCrawlMark.LookForMyCaller;
            RuntimeMethodHandleInternal methodHandle = InternalGetCurrentMethod(new StackCrawlMarkHandle(ref stackMark));
            return methodHandle.IsNullHandle() ? null : RuntimeType.GetMethodBase(null, methodHandle);
        }
```

**Comment by AustinWise** [comment]
I was able to reproduce the regression in [a benchmark](https://github.com/AustinWise/attachments/tree/master/dotnet/runtime/99852). Interestingly, it appears that something between .NET 9 Preview 2 and the merge base sped things up: 


| dotnet/runtime | Mean     | Error   | StdDev  |
|----------------- |---------:|--------:|--------:|
| .NET 9 Preview 2 | 698.8 ns | 2.27 ns | 2.12 ns |
| This PR | 639.9 ns | 3.62 ns | 3.38 ns |
| Merge Base | 571.2 ns | 2.98 ns | 2.79 ns |

Anyways, I'll mark this as draft while I fix the regression.

**Comment by AustinWise** [comment]
@jkotas Thanks for the tip, this PR is now a performance improvement.

| code           | Mean     | Error   | StdDev  |
|----------------- |---------:|--------:|--------:|
| this PR | 480.5 ns | 6.14 ns | 5.13 ns |
| merge base | 571.2 ns | 2.98 ns | 2.79 ns |
| .NET 9 preview 2 | 698.8 ns | 2.27 ns | 2.12 ns |

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**